python/ray/autoscaler/_private/load_metrics.py

```diff
@@ -138,23 +141,24 @@ def prune_active_ips(self, active_ips):
         active_ips.add(self.local_ip)
 
         def prune(mapping, should_log):
-            unwanted = set(mapping) - active_ips
-            for unwanted_key in unwanted:
+            unwanted_ips = set(mapping) - active_ips
+            for unwanted_ip in unwanted_ips:
                 if should_log:
-                    logger.info("LoadMetrics: "
-                                "Removed mapping: {} - {}".format(
-                                    unwanted_key, mapping[unwanted_key]))
-                del mapping[unwanted_key]
-            if unwanted and should_log:
-                # TODO (Alex): Change this back to info after #12138.
+                    logger.info(
+                        "LoadMetrics: "
+                        f"Removed ip: {unwanted_ip}."
+                        "\n(is not the ip of a node known to the autoscaler)")
```

Comment:
the message is not very clear (or maybe I am missing something?)

> (is not the ip of a node known to the autoscaler) 
what is "not the ip of a node" ?

Reply:
I think I'll remove the extra line and add an explanation in a comment in the code -- "The ip was read from the GCS but is not among the ips known to the autoscaler's NodeProvider. "

Some refactoring along the lines of https://github.com/ray-project/ray/issues/19086 could help.

Reply:
Added docstring + comment.